sklearn_numba_dpex/kmeans/kernels/compute_euclidean_distances.py

```diff
@@ -82,32 +84,35 @@ def compute_distances(
 
             for _1 in range(n_windows_for_features):
                 is_last_feature_window = _1 == last_feature_window_idx
+
                 load_window_of_centroids_and_features(
                     first_feature_idx,
                     loading_centroid_idx,
                     window_loading_centroid_idx,
                     window_loading_feature_offset,
                     current_centroids_t,
                     centroids_window,
-                    is_last_feature_window
                 )
 
                 dpex.barrier(dpex.CLK_LOCAL_MEM_FENCE)
-                accumulate_sq_distances(sample_idx, first_feature_idx, X_t, centroids_window, sq_distances, is_last_feature_window, is_last_centroid_window)
 
-                dpex.barrier(dpex.CLK_LOCAL_MEM_FENCE)
+                accumulate_sq_distances(
+                    sample_idx, first_feature_idx, X_t, centroids_window, sq_distances, is_last_feature_window, is_last_centroid_window
+                )
 
                 first_feature_idx += centroids_window_height
 
+                dpex.barrier(dpex.CLK_LOCAL_MEM_FENCE)
```

Comment:
Are there any particular motivation for moving barrier in kernels, such as this one?

Reply:
I find it more sensible to set the barrier to the latest possible instruction (it maybe gives more room for the threads to synchronize) but in fact I don't think it really matters